### ISSUE TITLE
[BUG]: iOS Audio Ducking on muted or silent video 

### ISSUE THREAD

**Issue by ThomasMoran** [TheWidlarzGroup/react-native-video#3528]
### What happened?

I have a video with no audio, which I do not want to duck outside audio (Spotify etc). For this reason I set `mixWithOthers={MixWithOthersType.MIX}` on the Video component. However, on the same screen I have another player which plays small voice prompts, and when they play I want to duck the outside audio.

The problem is that once I have set `mixWithOthers={MixWithOthersType.MIX}`, this will override the ducking on other audio providers, but if I set `mixWithOthers={MixWithOthersType.DUCK}`, then once the Video component is on screen, all audio will duck, even though my video is muted.

Do you think that when I set `mixWithOthers={MixWithOthersType.DUCK}` on my Video component, that this should not duck outside audio if the video itself is muted?

### Steps to reproduce

1. Add a `Video` with the `muted` property
2. Set `mixWithOthers={MixWithOthersType.DUCK}`
3. Have a 3rd party app playing sound
4. Load the screen with the muted video - the audio will duck

### react-native-video version

6.0.0-beta

### Platforms

iOS

### Architecture

Old Architecture

**Comment by WildSpy** [comment]
yes, I have the same thing, on iOS any videos (with or without sound) stop the sound of third-party applications. I tried to set ```mixWithOthers={MixWithOthersType.MIX}``` without any success

**Comment by WildSpy** [comment]
as it turned out, the options will be set only if there is also a category present, so `mixWithOthers` will not work without `ignoreSilentSwitch`

use this:

```js
mixWithOthers={MixWithOthersType.MIX}
ignoreSilentSwitch={IgnoreSilentSwitchType.OBEY}
```

fix this behavior or add it to docs please

![CleanShot 2024-02-26 at 14 28 26@2x](https://github.com/react-native-video/react-native-video/assets/1746527/bd10ca74-766d-48f6-90ac-5557c5a7714b)

**Comment by KrzysztofMoch** [comment]
@WildSpy thank you for your investigation 🙏
Does `ignoreSilentSwitch` need to be set to `OBEY` to make mixWithOthers work, or it can be other value ?

**Comment by WildSpy** [comment]
> @WildSpy thank you for your investigation 🙏 Does `ignoreSilentSwitch` need to be set to `OBEY` to make mixWithOthers work, or it can be other value ?

I think it can be any value, it just has to be, according to the condition in the screenshot above. `ignoreSilentSwitch` is `category`

**Comment by ThomasMoran** [comment]
Sorry guys, I stupidly put tags in my comment which blocked some of the text. 

My issue is slightly different. I can get mix and duck to work fine individually
What happens is that I have a muted video where I have to set MixWithOthersType.MIX on the Video component because otherwise it will DUCK by default, even though it is muted. Setting it to MIX will cause all audio in the app to MIX. I want other audio (using react-native-tts) to DUCK but it gets overridden by the MixWithOthersType.MIX on this library.

I want the Video component to not mess with the audio ducking/mixing at all whenever the muted property is set.

Not sure how possible this is

**Comment by TWG-AI** [comment]
Thanks for the detailed use case! This has been logged for the team.

The root issue here is that iOS's `AVAudioSession` is a process-wide singleton — setting `mixWithOthers={DUCK}` affects the entire app's audio session, not just the individual video instance. So a muted video can still activate audio ducking because it's an active AVPlayer instance regardless of its muted state.

A workaround: conditionally set `mixWithOthers` based on the `muted` prop. For the muted video, use `mixWithOthers={MixWithOthersType.MIX}`, and for the voice prompt player, use `{MixWithOthersType.DUCK}` — but note that the last-mounted player will "win" the audio session configuration.

Also, as @WildSpy noted in comments, `mixWithOthers` may only take effect when `ignoreSilentSwitch` is set as well — worth checking if that combination changes behavior for your use case.

> I'm an automated triage assistant — please verify the details here. A human maintainer will follow up.
